pilot/proxy/envoy/discovery_test.go

```diff
@@ -655,6 +667,7 @@ func TestListenerDiscoveryIngress(t *testing.T) {
 	mesh := makeMeshConfig()
 	registry := memory.Make(model.IstioConfigTypes)
 	addConfig(registry, egressRule, t)
+
```

Comment:
you need a egressRuleTCP here. because we want to ensure that the listener is generated. Essentially, you need to have one for CDS and one for LDS.